tests/test_server_basic.py

```diff
@@ -221,6 +220,55 @@ def test_valid_token(dispatcher_live_fixture):
     logger.info(json.dumps(jdata, indent=4))
 
 
+@pytest.mark.parametrize("instrument", ["", "None", None, "undefined"])
+@pytest.mark.parametrize("public_request", [True, False])
+def test_download_products(dispatcher_live_fixture, empty_products_files_fixture, instrument, public_request):
+    server = dispatcher_live_fixture
+
+    logger.info("constructed server: %s", server)
+
+    if public_request:
+        encoded_token = None
+    else:
+        # let's generate a valid token with high threshold
+        token_payload = dict(
+            **default_token_payload,
+            mstout=True,
+            mssub=True,
+            intsub=5
+        )
+
+        encoded_token = jwt.encode(token_payload, secret_key, algorithm='HS256')
+
+    session_id = empty_products_files_fixture['session_id']
+    job_id = empty_products_files_fixture['job_id']
+
+    params = {
+            'instrument': instrument,
+            # since we are passing a job_id
+            'query_status': 'ready',
+            'file_list': 'test.fits.gz.recovered',
+            'download_file_name': 'output_test',
+            'session_id': session_id,
+            'job_id': job_id,
+            'token': encoded_token
+        }
+
+    c = requests.get(server + "/download_products",
+                     params=params)
+
+    assert c.status_code == 200
+
+    # download the output, read it and then compare it
+    with gzip.open(f'scratch_sid_{session_id}_jid_{job_id}/output_test', 'wb') as fout:
+        fout.write(empty_products_files_fixture['content'])
+    data_downloaded = ''
+    with gzip.open(f'scratch_sid_{session_id}_jid_{job_id}/output_test', 'rb') as fout:
+        data_downloaded = fout.read()
+
+    assert data_downloaded == empty_products_files_fixture['content']
+
```

Comment:
you are writing `empty_products_files_fixture['content']` to a file, reading it back, and verifying that this did not change the content, right?

This process is not involving flask or dispatcher, so they are not tested.

Reply:
My bad. Adjusting